core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/MultipleChoiceInputCheckerServiceSpec.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for MultipleChoiceInputCheckerService.
+ */
+
+describe('Multiple choice input checker service', function() {
+  beforeEach(module('oppia'));
+
+  var AnswerGroupObjectFactory, mcics, customizationArgs,
+    answerGroupsFalse, answerGroupsTrue;
+
+  customizationArgs = {
+    choices: {
+        value: [
+          '<p>Option 1</p>',
+          '<p>Option 2</p>',
+          '<p>Op 3</p>'
+        ]
+      }
+  };
+
+  beforeEach(inject(function($injector) {
+    AnswerGroupObjectFactory = $injector.get('AnswerGroupObjectFactory');
+    mcics = $injector.get('MultipleChoiceInputCheckerService');
+  }));
+
+  it('should return false for two Equals rules', function() {
+    answerGroupsFalse = [
```

Comment:
It doesn't look like there's a reason for this and answerGroupsTrue to be global, so you can remove them from line 22 and declare them with var here. Only declare them globally if multiple tests will use them.

Also, I suggest condensing it a bit and changing the name:

```
    invalidAnswerGroups = [{
      rules: [{
        type: 'Equals',
        inputs: {
          x: 0
        }
      }, {
        type: 'Equals',
        inputs: {
          x: 2
        }
      }],
      outcome: {
        param_changes: [],
        feedback: [],
        dest: 'Question 1'
      },
      correct: false
    }];
```

Ditto below.